### PR TITLE
Add new functions is_not_null and is_not_in

### LINKED ISSUE #541: Add negated variants of common methods in ehrQL
Negation can sometimes be a bit awkward in ehrQL as it involves the slightly cryptic `~` symbol and has to go at the beginning of the method chain.

I think it would be helpful to at least add `is_not_null()` and `is_not_in()` methods. Possibly there are others but they don't immediately spring to mind.

### PR DESCRIPTION
**Pull request by milanwiedemann**
Closes #541

### CODE REVIEW

File: tests/spec/code_series_ops/test_containment.py [hunk 1]
```diff
@@ -51,3 +64,27 @@ def test_is_in_codelist_csv(spec_test):
             4: None,
         },
     )
+
+
+def test_is_not_in_codelist_csv(spec_test):
```
Comment (evansd) [inline tests/spec/code_series_ops/test_containment.py:69]: I don't actually think we need this test. I think the first one you added does enough to specify the behaviour of `is_not_in` with respect to `is_in` and everything else follows from that.

**Review by evansd — APPROVED** [review]
Great, thanks for this @milanwiedemann 

I've made a couple of stylistic suggestions, and I also think we could do without one of the tests. But everything here is correct.

And just FYI, if you edit the PR description to include the line `Closes #541` then merging the PR will automatically close the associated ticket.

File: databuilder/query_language.py [hunk 2]
```diff
@@ -66,13 +66,23 @@ def __ne__(self, other):
     def is_null(self):
         return _apply(qm.Function.IsNull, self)
 
+    def is_not_null(self):
+        return ~_apply(qm.Function.IsNull, self)
```
[the remark this answers is not in the document]
Comment (milanwiedemann) [reply databuilder/query_language.py:70]: thanks! I see what you mean

File: databuilder/query_language.py [hunk 3]
```diff
@@ -66,13 +66,23 @@ def __ne__(self, other):
...
+
     def is_in(self, other):
         # The query model requires an immutable Set type for containment queries, but
         # that's a bit user-unfriendly so we accept other types here and convert them
         if isinstance(other, (tuple, list, set)):
             other = frozenset(other)
         return _apply(qm.Function.In, self, other)
 
+    def is_not_in(self, other):
```
[the remark this answers is not in the document]
Comment (milanwiedemann) [reply databuilder/query_language.py:79]: yes, your point is even clearer here :)

File: tests/spec/code_series_ops/test_containment.py (hunk 1 above)
Comment (milanwiedemann) [reply tests/spec/code_series_ops/test_containment.py:69]: thx!

**Comment by evansd** [comment]
@milanwiedemann It's in! The first of many ...

**Pull request merged**